examples/pipelines/golang-project/scanner.go

```diff
@@ -0,0 +1,256 @@
+	if err := g.runGoSec(ctx); err != nil {
+		return nil, fmt.Errorf("could not run gosec: %w", err)
+	}
+
+	vulns, err := g.parseVulns(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("could not parse vulns: %w", err)
+	}
+
+	return vulns, nil
+}
+
+func (g *goSecScanner) parseVulns(ctx context.Context) ([]*ocsf.VulnerabilityFinding, error) {
+	f, err := os.Open(goSecOutPath)
+	if err != nil {
+		return nil, fmt.Errorf("could not open gosec_out.json: %w", err)
+	}
+
+	defer func() {
+		if err := f.Close(); err != nil {
+			component.
+				LoggerFromContext(ctx).
+				Error(
+					"could not close gosec_out.json",
+					slog.String("err", err.Error()),
+				)
+		}
+
+		if err := os.RemoveAll(goSecOutPath); err != nil {
+			component.
+				LoggerFromContext(ctx).
+				Error(
+					"could not remove gosec_out.json",
+					slog.String("err", err.Error()),
+				)
+		}
+	}()
+
+	b, err := io.ReadAll(f)
+	if err != nil {
+		return nil, fmt.Errorf("could not read gosec_out.json: %w", err)
+	}
+
+	var out GoSecOut
+	if err := json.Unmarshal(b, &out); err != nil {
+		return nil, fmt.Errorf("could not decode gosec_out.json: %w", err)
+	}
+
+	var (
+		vulns = make([]*ocsf.VulnerabilityFinding, 0, len(out.Issues))
+		now   = time.Now().Unix()
+	)
+
+	for _, issue := range out.Issues {
+		vulns = append(vulns, &ocsf.VulnerabilityFinding{
+			ActivityId:   ocsf.VulnerabilityFinding_ACTIVITY_ID_CREATE,
+			CategoryUid:  ocsf.VulnerabilityFinding_CATEGORY_UID_FINDINGS,
+			ClassUid:     ocsf.VulnerabilityFinding_CLASS_UID_VULNERABILITY_FINDING,
+			Confidence:   &issue.Confidence,
+			ConfidenceId: confidences[issue.Confidence],
+			Count:        ptr(int32(1)),
+			FindingInfo: &ocsf.FindingInfo{
+				CreatedTime: &now,
+				DataSources: []string{
+					issue.File,
+				},
+				Desc:          &issue.Details,
+				FirstSeenTime: &now,
+				LastSeenTime:  &now,
+				ModifiedTime:  &now,
+				ProductUid:    ptr("gosec"),
+				Title:         issue.Details,
+				Uid:           issue.RuleID,
+			},
+			Message: ptr(issue.Details),
+			Resource: &ocsf.ResourceDetails{
+				Uid: ptr(
+					strings.Join([]string{
+						issue.File,
+						issue.Line,
+						issue.Column,
+					},
+						":",
+					),
+				),
+				Data: &structpb.Value{
+					Kind: &structpb.Value_StringValue{
+						StringValue: issue.Code,
+					},
+				},
+			},
+			RawData:    ptr(string(b)),
+			Severity:   &issue.Severity,
+			SeverityId: severities[issue.Severity],
+			StartTime:  &now,
+			Status:     ptr("opened"),
+			Time:       now,
+			TypeUid: int64(
+				ocsf.VulnerabilityFinding_CLASS_UID_VULNERABILITY_FINDING.Number()*
+					100 +
+					ocsf.VulnerabilityFinding_ACTIVITY_ID_CREATE.Number(),
+			),
+			Vulnerabilities: []*ocsf.Vulnerability{
+				{
+					Cwe: &ocsf.Cwe{
+						Uid:    issue.Cwe.ID,
+						SrcUrl: &issue.Cwe.URL,
+					},
+				},
+			},
+		})
+	}
+
+	return vulns, nil
+}
+
+func (g *goSecScanner) runGoSec(ctx context.Context) error {
```

Comment:
Using dockertest here just for the POC - we'll use gosec as a container in a real world scenario